tests/acceptance/blocking_adapter_test.py

```diff
@@ -476,13 +469,13 @@ def test(self):
         self.assertIsInstance(method, pika.spec.Basic.GetOk)
         self.assertEqual(method.delivery_tag, 1)
         self.assertFalse(method.redelivered)
-        self.assertEqual(method.exchange, _p3_as_bytes(''))
-        self.assertEqual(method.routing_key, q_name)
+        self.assertEqual(method.exchange, as_bytes(''))
+        self.assertEqual(method.routing_key, as_bytes(q_name))
```

Comment:
Naturally all such ideas should be backwards compatible, which is easy, since no python3 support and text vs. binary data issues have existed (sort of), since in python2 people usually pass around mystery binary blobs. In such environment py2 would continue to handle short strings and long strings as binary data and py3 would handle text to data conversions.